### ISSUE TITLE
Add an i386 pipeline to continuous integration

### ISSUE THREAD

**Issue by Minoru** [newsboat/newsboat#556]
2.16 failed to build on i386, and we never knew since we don't have it on Travis. Need to add this architecture to the pipeline, to avoid such breakage in the future.

**Comment by Minoru** [comment]
Can't find any confirmation or a knob for i386 on Travis. Need to look into other CI clouds:
- [Cirrus](https://cirrus-ci.org/) (which we're going to use for FreeBSD #407)
- [Drone](https://drone.io/)
- [Circle](https://circleci.com/)
- [AppVeyor](https://www.appveyor.com/)

**Comment by Minoru** [comment]
Travis apparently doesn't add i386 architecture by default: https://travis-ci.org/newsboat/newsboat/jobs/552424973

Cirrus, Drone, and Circle clouds don't seem to have i386.

AppVeyor mentions "x86 platform", I think I'll try that next. If that fails, the next option is to double down on multi-arch Ubuntu, install i386 versions of all our deps, pass additional flags to GCC and Rustc, and cross-compile that way.

**Comment by skellat** [comment]
How much of an i386 installed base is there for newsboat to operate on?

**Comment by Minoru** [comment]
Surprisingly hard to find data on that. The best I could muster is the number of submissions to Debian Popularity Contest, which says there are around 25,000 i386 machines: https://popcon.debian.org/stat/sub-i386.png On the other hand, amd64 has 180,000: https://popcon.debian.org/stat/sub-amd64.png There are 250,000 participants in total, meaning every 10th participant is running i386. I'd say this is a significant installed base.

It's another question if Newsboat is even usable on 32-bit—i.e. memory-constrained—machines. We have a memory leak still, which limits how you can use Newsboat if you have only 4Gb of RAM.

The main reason I want to add i386 to CI is not to be a jerk. There is no reason to break i386 compatibility at the moment, so let's not break it.

**Comment by Minoru** [comment]
On AppVeyor, [`platform: x86` relates to some msbuild stuff, not host architecture](https://help.appveyor.com/discussions/questions/213-platform-x86-is-not-32-bit). I'm looking into cross-compilation now.

**Comment by Minoru** [comment]
Got cross-compilation working in a Qemu VM. Far easier than I expected; I almost regret spending time on CI clouds now :) Tomorrow I will work on turning this into a Travis CI job.

**Comment by Minoru** [comment]
Our i386 job needs a completely separate set of packages from the other jobs. We currently use two global `apt` sections; I'd have to move them somewhere else, and then play with YAML references to make all the relevant jobs get a copy.

This looked like a boring task, and I was already impressed with Cirrus CI after setting up a FreeBSD job there. So I created an i386 job there, not on Travis.

I'll need to test i386 locally in the future, so I also invested in another thing that's been on my to-do list forever: Docker. The i386 job is completely dockerized, meaning I (or anyone else) can have the same environment both locally and on CI.

PR with all that goodness is coming shortly.
